esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -0,0 +1,205 @@
+"""Fixes that are shared between datasets and drivers."""
+import logging
+from cf_units import Unit
+import cordex as cx
+import numpy as np
+import iris
+
+from iris.coord_systems import RotatedGeogCS, LambertConformal
+from esmvalcore.cmor.fix import Fix
+from esmvalcore.exceptions import RecipeError
+logger = logging.getLogger(__name__)
+
+
+class MOHCHadREM3GA705(Fix):
+    """General fix for MOHC-HadREM3-GA7-05."""
+
+    def fix_metadata(self, cubes):
+        """Fix time long_name, and latitude and longitude var_name.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            cube.coord('latitude').var_name = 'lat'
+            cube.coord('longitude').var_name = 'lon'
+            cube.coord('time').long_name = 'time'
+
+        return cubes
+
+
+class TimeLongName(Fix):
+    """Fixes for time coordinate."""
+
+    def fix_metadata(self, cubes):
+        """
+        Fix time long_name.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            cube.coord('time').long_name = 'time'
+
+        return cubes
+
+
+class CLMcomCCLM4817(Fix):
+    """Fixes for CLMcom-CCLM4-8-17."""
+
+    def fix_metadata(self, cubes):
+        """Fix calendars.
+
+        Set calendar to 'proleptic_gregorian' to avoid
+        concatenation issues between historical and
+        scenario runs.
+
+        Fix dtype value of coordinates and coordinate bounds.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            if cube.coord('time').units.calendar == 'standard':
+                cube.coord('time').units = Unit(
+                    'days since 1850-1-1 00:00:00',
+                    calendar='proleptic_gregorian'
+                )
```

Comment:
Because some files had standard calendars and some other proleptic gregorian and that broke the concatenation. The reference date was the same in that case, yes.

Reply:
How about using the [`change_calendar`](https://cf-units.readthedocs.io/en/latest/unit.html#cf_units.Unit.change_calendar) method, would that be safer?

Reply:
Done, I did not know about change_calendar, that will be useful!